python/mxnet/text/embedding.py

```diff
@@ -0,0 +1,722 @@
+        ...     def __init__(self, pretrain_file='my_pretrain_file'):
+        ...         pass
+        >>> embed = mxnet.text.embedding.TextEmbed.create('MyTextEmbed')
+        >>> print(type(embed))
+        <class '__main__.MyTextEmbed'>
+        """
+
+        assert(isinstance(embed_cls, type))
+        embed_name = embed_cls.__name__.lower()
+        if embed_name in TextEmbed.embed_registry:
+            warnings.warn('New embedding %s.%s is overriding existing '
+                          'embedding %s.%s', embed_cls.__module__,
+                          embed_cls.__name__,
+                          TextEmbed.embed_registry[embed_name].__module__,
+                          TextEmbed.embed_registry[embed_name].__name__)
+        TextEmbed.embed_registry[embed_name] = embed_cls
+        return embed_cls
+
+    @staticmethod
+    def create(embed_name, **kwargs):
+        """Creates an instance of :func:`~mxnet.text.embedding.TextEmbed`.
+
+        Creates a text embedding instance by loading embedding vectors from an
+        externally hosted pre-trained text embedding file, such as those
+        of GloVe and FastText. To get all the valid `embed_name` and
+        `pretrain_file`, use
+        `mxnet.text.embedding.TextEmbed.get_embed_names_and_pretrain_files()`.
+
+
+        Parameters
+        ----------
+        embed_name : str
+            The text embedding name (case-insensitive).
+
+
+        Returns
+        -------
+        mxnet.text.glossary.TextEmbed:
+            A text embedding instance that loads embedding vectors from an
+            externally hosted pre-trained text embedding file.
+        """
+        if embed_name.lower() in TextEmbed.embed_registry:
+            return TextEmbed.embed_registry[embed_name.lower()](**kwargs)
+        else:
+            raise ValueError('Cannot find embedding %s. Valid embedding '
+                             'names: %s' %
+                             (embed_name,
+                              ', '.join(TextEmbed.embed_registry.keys())))
+
+    @staticmethod
+    def check_pretrain_files(pretrain_file, embed_name):
+        """Checks if a pre-trained text embedding file name is valid.
+
+
+        Parameters
+        ----------
+        pretrain_file : str
+            The pre-trained text embeddibg file.
+        embed_name : str
+            The text embedding name (case-insensitive).
+        """
+        embed_name = embed_name.lower()
+        embed_cls = TextEmbed.embed_registry[embed_name]
+        if pretrain_file not in embed_cls.pretrain_file_sha1:
+            raise KeyError('Cannot find pretrain file %s for embedding %s. '
+                           'Valid pretrain files for embedding %s: %s' %
+                           (pretrain_file, embed_name, embed_name,
+                            ', '.join(embed_cls.pretrain_file_sha1.keys())))
+
+    @staticmethod
+    def get_embed_names_and_pretrain_files():
```

Comment:
list_pretrained_embeddings?

Reply:
Thanks. It returns a string rather than prints the string. Thus I guess "get_" is better than "list_"